### PR TITLE
publish artifacts to Marketplace instead of docker

### PR DESCRIPTION
**Pull request by nullable-eth**
### Description of your changes
Update build infra to publish new build images to Marketplace instead of docker hub.

Fixes #[468](https://github.com/upbound/shimmer/issues/468)

I have:

- [X] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [X] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested
Hasn't, tried to run workflow with `act workflow_dispatch -P ubuntu-22.04=catthehacker/ubuntu:act-22.04 -W .github/workflows/ci.yml --insecure-secrets --secret-file secrets`, can create upbound robot secrets locally but still missing production secrets that can't retrieve locally so it won't run.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -10,14 +10,14 @@ on:
...
-  GO_VERSION: '1.19'
+  GO_VERSION: '1.19.4'
   GOLANGCI_VERSION: 'v1.49'
   DOCKER_BUILDX_VERSION: 'v0.8.2'
 
   # Common users. We can't run a step 'if secrets.AWS_USR != ""' but we can run
   # a step 'if env.AWS_USR' != ""', so we copy these to succinctly test whether
   # credentials have been provided before trying to run steps that need them.
-  DOCKER_USR: ${{ secrets.DOCKER_USR }}
+  UPBOUND_ROBOT_ID: ${{ secrets.UPBOUND_ROBOT_ID }}
```
Comment (jastang) [inline .github/workflows/ci.yml:20]: Not picky about names, but perhaps for consistency other repos publishing to Marketplace use `XPKG_ACCESS_ID` and `XPKG_TOKEN` , e.g. [provider-aws](https://github.com/crossplane-contrib/provider-aws/blob/master/.github/workflows/ci.yml#L309-L311)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -234,25 +234,26 @@ jobs:
           name: output
           path: _output/**
 
-      - name: Login to Docker
+      - name: Login to Marketplace
```
Comment (jastang) [inline .github/workflows/ci.yml:237]:
```suggestion
      - name: Login to Upbound
```

Comment (jastang) [reply .github/workflows/ci.yml:237]: suggesting r/Marketplace/Upbound on this file; feels more precise as "Marketplace" could be ambiguous

**Review by jastang — COMMENTED** [review]
Minor suggestions but looks good so far sans the CI setup on Upbound + GH. Do we need to backfill/index versions on Dockerhub into xpkg.upbound.io?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nullable-eth) [reply .github/workflows/ci.yml:20]: good suggestion, done.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (nullable-eth) [reply .github/workflows/ci.yml]: What about Upbound Marketplace?  More specific and more branded.

File: Makefile [hunk 3]
```diff
@@ -78,7 +78,7 @@ HELM_CHART_LINT_ARGS_$(PACKAGE_NAME) = --set nameOverride='',imagePullSecrets=''
 # Due to the way that the shared build logic works, images should
 # all be in folders at the same level (no additional levels of nesting).
 
-REGISTRY_ORGS ?= docker.io/upbound
+REGISTRY_ORGS ?= xpkg.upbound.io/upbound
```
Comment (hasheddan) [inline Makefile:81]: Will also need to update location in https://github.com/upbound/universal-crossplane/blob/2b692a94243814661df73be4ca9212f244c4d824/cluster/charts/universal-crossplane/values.yaml.tmpl#L157

Comment (nullable-eth) [reply Makefile:81]: Done, thanks!

**Comment by nullable-eth** [comment]
This GTG except the PROD robot token (with access to the already created universal-crossplane and uxp-bootstrapper PROD repo) and corresponding github secrets (which I do not have access to on this repository).  The no index rules are live and in PROD.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (jastang) [reply .github/workflows/ci.yml]: Hmm, on second thought these aren't going to be xpkgs by definition, so that wouldn't be an accurate name.

Comment (nullable-eth) [reply .github/workflows/ci.yml]: But it is interacting with the Marketplace OCI URL which is xpkg.upbound.io so it's at least half appropriate.

Comment (jastang) [reply .github/workflows/ci.yml]: Fair point 👍

**Review by hasheddan — APPROVED** [review]
Thanks @nullable-eth ! I have added the required secrets 👍🏻

**Pull request merged**